runtime/warp.cu

```diff
@@ -161,4 +161,93 @@ __device__ void warpReduceTIDXY(
     reduction_op(out, reduce_val);
   }
 }
+
+// sizeof(T) * K = sizeof(uint64_t)
+// require alginment of sizeof(T) * K to safely cast between T and uint64_t
+// shfl uses uint64_t to reduce number of shuffles
```

Comment:
nitpick question: since this is runtime file, I don't think we can do concepts there, can we?

Reply:
didn't check that, but `// Array structure ensures data is aligned for safe casting to uint64_t`. So we actually don't need to add extra checks.